include/dlaf/multiplication/general/impl.h

```diff
@@ -19,13 +19,38 @@
 #include <dlaf/communication/communicator_grid.h>
 #include <dlaf/matrix/distribution.h>
 #include <dlaf/matrix/index.h>
+#include <dlaf/matrix/matrix.h>
+#include <dlaf/matrix/matrix_ref.h>
 #include <dlaf/matrix/panel.h>
 #include <dlaf/multiplication/general/api.h>
 #include <dlaf/sender/when_all_lift.h>
+#include <dlaf/util_matrix.h>
 
 namespace dlaf::multiplication {
 namespace internal {
 
+template <Backend B, Device D, class T>
+void General<B, D, T>::callNN(const T alpha, MatrixRef<const T, D>& mat_a, MatrixRef<const T, D>& mat_b,
+                              const T beta, MatrixRef<T, D>& mat_c) {
+  namespace ex = pika::execution::experimental;
+
+  DLAF_ASSERT_HEAVY(matrix::multipliable(mat_a, mat_b, mat_c, blas::Op::NoTrans, blas::Op::NoTrans),
+                    mat_a, mat_b, mat_c);
+
+  for (SizeType j = 0; j < mat_c.nrTiles().cols(); ++j) {
+    for (SizeType i = 0; i < mat_c.nrTiles().rows(); ++i) {
+      for (SizeType k = 0; k < mat_a.nrTiles().cols(); ++k) {
```

Comment:
The case `k=0` is not handled correctly.
C should be scaled according to `beta`.